src/components/core/geoChart/tests/geoChartColor.spec.ts

```diff
@@ -38,7 +38,13 @@ function createColorStrategy(
     };
 
     if (segmentItems && segmentItems.length) {
-        const items = uniq(segmentItems).map((name, index) => {
+        const segmentItemsUniq = segmentItems.reduce((result, item) => {
+            const name = item || EMPTY_SEGMENT_ITEM;
+            result.push(name);
+            return uniq(result);
```

Comment:
You run `uniq` for every single loop in  `reduce` which is not a good practise IMO.
How about
```
        const segmentItemsWithoutNull: string[] = segmentItems.map(
            (name: string): string => name || EMPTY_SEGMENT_ITEM,
        );
        const items = uniq(segmentItemsWithoutNull).map((name, index) => ({
            attributeHeaderItem: {
                name,
                uri: `/gdc/md/projectId/obj/2/elements?id=${index}`,
            },
        }));
```

Reply:
fixed